src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/XmlFormatReaderGenerator.cs

```diff
@@ -98,17 +98,19 @@ private class CriticalHelper
             private ArgBuilder _memberNamesArg;
             private ArgBuilder _memberNamespacesArg;
             private ArgBuilder _collectionContractArg;
-
-            private bool _useReflection = true;
+#endif
 
             public XmlFormatClassReaderDelegate GenerateClassReader(ClassDataContract classContract)
             {
-                if (_useReflection)
+                if (DataContractSerializer.Option == SerializationOption.ReflectionAsBackup || DataContractSerializer.Option == SerializationOption.ReflectionOnly)
                 {
```

Comment:
Per our previous discussion, we'd like to set ReflectionAsBackup as the default option. The 'else' block not being hit when ReflectionAsBackup, however, means CoreCLR doesn't work in that mode. 

---
In reply to: [63756585](https://github.com/dotnet/corefx/pull/8629#discussion_r63756585) [](ancestors = 63756585,63755899)